x-pack/test/api_integration/deployment_agnostic/apis/observability/dataset_quality/degraded_docs.ts

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { log, timerange } from '@kbn/apm-synthtrace-client';
+import expect from '@kbn/expect';
+import { SupertestWithRoleScopeType } from '../../../services';
+import { DeploymentAgnosticFtrProviderContext } from '../../../ftr_provider_context';
+import {
+  DatasetQualitySupertestUser,
+  getDatasetQualityMonitorSupertestUser,
+  getDatasetQualityNoAccessSuperTestUser,
+} from './utils';
+
+export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const synthtrace = getService('logsSynthtraceEsClient');
+  const start = '2023-12-11T18:00:00.000Z';
+  const end = '2023-12-11T18:01:00.000Z';
+
+  async function callApiAs(roleScopedSupertestWithCookieCredentials: SupertestWithRoleScopeType) {
+    return roleScopedSupertestWithCookieCredentials
+      .get(`/internal/dataset_quality/data_streams/degraded_docs`)
+      .query({
+        type: 'logs',
+        start,
+        end,
+      });
+  }
+
+  describe('Degraded docs', function () {
+    describe('Authorization', function () {
+      let supertestNoAccessUser: DatasetQualitySupertestUser;
+      before(async function () {
+        supertestNoAccessUser = await getDatasetQualityNoAccessSuperTestUser({
+          getService,
+        });
+
+        if (!supertestNoAccessUser.isCustomRoleEnabled) {
+          this.skip(); // The test is only relevant when custom roles are enabled
+        }
```

Comment:
This is definitely anti-pattern in terms of deployment-agnostic tests.

Custom roles are not available for Observability serverless project, so this test should not be transfered to deployment-agnostic scope at the moment.

Please check the docs https://github.com/elastic/kibana/tree/main/x-pack/test/api_integration/deployment_agnostic#when-to-create-separate-tests

Reply:
Please move serverless-specific tests under `x-pack/test_serverless`, feature flag config for Oblt project has custom roles enabled. https://github.com/elastic/kibana/blob/main/x-pack/test_serverless/api_integration/test_suites/observability/config.feature_flags.ts#L25

Until custom roles are enabled this test shouldn't be part of deployment-agnostic suite.

Reply:
Please help me understand why do we need a custom role here?
couldn't we work with `viewer`, `editor` and/or `admin`?
For specific roles (e.g. just access to `logs-apm*-*` vs `logs-*-*`) we could hold the tests until custom roles are supported (and keep them in their own folders), but for the rest of the functionality why do we need them?

Reply:
In stateful we had the custom role which most of the tests ran against: 
```
  [DatasetQualityCustomRolename.datasetQualityMonitorUser]: {
    elasticsearch: {
      indices: [
        {
          names: ['logs-*-*', 'metrics-*-*', 'traces-*-*', 'synthetics-*-*'],
          privileges: ['monitor', 'view_index_metadata'],
        },
      ],
    },
  },
```
But I am fine going with the predefined roles in deployment agnostic.

Reply:
yes, but I think most of those privileges are granted to at least `editor` role by default in serverless, not sure about `viewer`